workshops/filters.py

```
@@ -207,6 +211,139 @@ def get_order_by(self, order_value):
+        widget=widgets.CheckboxInput)
+
+    is_swc_instructor = django_filters.BooleanFilter(
+        label='Is SWC Instructor?',
+        name='is_swc_instructor',
+    )
+
+    is_dc_instructor = django_filters.BooleanFilter(
+        label='Is DC Instructor?',
+        name='is_dc_instructor',
+    )
+
+    non_evaluated_homework = django_filters.BooleanFilter(
+        label='Unevaluated homework?',
+        action=filter_unevaluated_homework,
+    )
+
+    training = django_filters.ModelChoiceFilter(
+        queryset=Event.objects.ttt(),
+        action=filter_training,
+    )
+
+    class Meta:
+        model = Person
+        fields = [
+            'search',
+            'all_persons',
+            'is_swc_instructor',
+            'is_dc_instructor',
+            'training',
+        ]
+        order_by = ["-last_login", "lastname", "-lastname", "firstname", "-firstname",
+                    "email", "-email"]
+
+    def get_order_by(self, order_value):
+        if order_value == 'firstname':
+            return ['personal', 'middle', 'family']
+        elif order_value == '-firstname':
+            return ['-personal', '-middle', '-family']
+        elif order_value == 'lastname':
+            return ['family', 'middle', 'personal']
+        elif order_value == '-lastname':
+            return ['-family', '-middle', '-personal']
+        else:
+            return super().get_order_by(order_value)
+
+
+def filter_matched(queryset, choice):
+    if choice == '':
+        return queryset
+    elif choice == 'u':  # unmatched
+        return queryset.filter(person=None)
+    elif choice == 'p':  # matched person, unmatched training
+        return queryset.filter(person__isnull=False)\
+                       .exclude(person__task__role__name='learner',
+                                person__task__event__tags__name='TTT')\
+                       .distinct()
+    else:
+        assert choice == 't'  # matched person and training
```

Part of contract of this function is that `choice` must be `''`, `'u'`, `'p'` or `'t'`. We want to fail loud when `choice` is something different.